server/container/guice/jpa-common-guice/src/main/java/org/apache/james/modules/data/JPAConfiguration.java

```diff
@@ -18,67 +18,131 @@
  ****************************************************************/
 package org.apache.james.modules.data;
 
-import org.apache.james.backends.jpa.JPAConstants;
+import java.util.Optional;
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
 
 public class JPAConfiguration {
 
-    public static Builder builder() {
-        return new Builder();
+    public static class Credential {
+        public static Optional<Credential> of(String username, String password) {
+            try {
+                return Optional.of(new Credential(username, password));
+            } catch (IllegalArgumentException e) {
+                return Optional.empty();
```

Comment:
Why don't we just let this exception be returned to the user rather than doing some sanitizing?